client/state/sites/actions.js

```diff
@@ -231,6 +231,20 @@ export function deleteSite( siteId ) {
 					return;
 				}
 
+				if ( error.error === 'unauthorized' ) {
+					// this error is used when you try to delete an Atomic site
+					// right after removing the plan(s) from it. Normally the
+					// message will be 'User cannot delete jetpack site via API.'
+					// The user needs to wait a minute before self-service deletion is available.
```

Comment:
How could we prevent this from happening? Maybe after removing the plans, we could force-sync the permissions or something?

Reply:
is there a way to determine if delete is ready? we could disable the button and add some info about it being ready in 5 mins or so.

Reply:
@zaguiini We have a feedback flow for removing the plan, which might capture the same feedback. If they remove the plan just before removing the site, it could be a bit much to ask for two lots. 

@zaguiini @vykes-mac, I agree this is not optimal. From memory, I think there is an async job that runs when you remove an Atomic plan, and this needs some time to complete. I didn't check if it sets some flag somewhere to say "in progress" which we could expose via the API and use on the frontend. That could be a nice follow-up PR.

Reply:
@mpkelly I tested this flow to see what differences are in the site's object in those three states: site before removal, site after removing the plan, and site eligible for removal after the async job runs, but I've got stuck. I waited 15 minutes, and I'm still getting the "Unable to delete site. If you recently removed the plan(s) then please wait a couple of minutes and retry." error message. Should this process work for a site which is still Atomic?

Reply:
@wojtekn, thanks for testing this. 

>Should this process work for a site which is still Atomic?

On trying to delete a site that is still Atomic, i.e. a site with a Business plan, it should tell you to remove the plan. This was the existing logic. After removing the plan, if you retry deleting the site, it should call the API to try and do that. Depending on how long you waited, it should either be successful or show you the notice that I added. 

What is the current state of the site you were trying to delete? Does it still have an Atomic domain with "staging" in it? Does it still have plans shown? It sounds like the async job might have not been completed.

Reply:
Thanks @mpkelly . I could test this flow now by triggering plan removal from Calypso.

I think I see a cleaner way of warning the user that their site is not ready for removal yet. Now, when they click the "Delete site" button for the site that still has paid upgrade, we display this message in popup:

<img width="669" alt="Screen Shot 2022-12-22 at 10 32 33" src="https://user-images.githubusercontent.com/727413/209104415-b4e280bf-3915-41f0-b628-5a4ec9429107.png">

When they remove the paid upgrades, the site still passes the `isAtomic` checks, but has a `free_plan` set. For that case, we could introduce another popup saying something like _"Your site is not ready for removal yet - we are currently reverting it. If you recently removed the plan(s) then please wait a couple of minutes before you try removing the site."_

That way, we would show the final confirmation popup with the "Confirm delete site" heading and the input field only if the site is really applicable for removal.

Reply:
Thanks, @wojtekn. That sounds like an improvement. I will investigate and update the PR.

Reply:
I've tested this flow twice, and it's looking good. It will save many support interactions.
The first time I didn't need to wait, and I deleted the site using the wpcomstaging subdomain. Very smooth process, great job 👍 .


https://user-images.githubusercontent.com/779993/209147840-dff7551c-bd64-41ce-9de2-9ce2063803e0.mp4


The second time I waited a bit longer, I got an error loading the dashboard due to the subdomain migration from wpcomstaging.com to wordpress.com. This PR does not produce this issue, but it is something to consider.

<img width="873" alt="Screenshot 2022-12-22 at 13 47 33" src="https://user-images.githubusercontent.com/779993/209147827-00ee171d-1138-4053-9d1b-fea43ef8740c.png">


My only suggestion that we can address in a separate PR is adding a link to settings in the Removed plan notice.
It will be easier for the user to return to the settings and continue deleting the site with fewer clicks.
<img width="679" alt="Screenshot 2022-12-22 at 13 22 12" src="https://user-images.githubusercontent.com/779993/209145533-517a1141-e4b7-4e7c-a9a5-69797c00605f.png">

Reply:
Thanks, @mpkelly. It's better now, as we inform the user upfront instead of relying on the API error.

One thing still bothers me - we block delete action in two cases: when the site still has a paid upgrade and when it is not eligible for removal because it's still Atomic. In both cases, we use different ways of displaying the message: in the first, we use the popup. In the second, we display that new text above the button, and we disable the button. Would it be better to use the same way in both cases?